### PR TITLE
Add `--zap ZAP` to `run_zaptool.sh` to Specify an Arbitrary Location to `zap`

### PR DESCRIPTION
**Pull request by gerickson**
Add `--zap ZAP` to `run_zaptool.sh` to Specify an Arbitrary Location to `zap` (#35549).

* _scripts/tools/zap/run_zaptool.sh_
    * Display the value of `ZAP_CMD` to standard output.
    * Add support for the `--zap` command line option to specify the `zap` executable to run.
    * Added usage documentation about `ZAP_DEVELOPMENT_PATH` and `ZAP_INSTALL_PATH`.
* _docs/code_generation.md_: Added documentation about the `--zap ZAP` option to `run_zaptool.sh`.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: scripts/tools/zap/run_zaptool.sh [hunk 1]
```diff
@@ -39,6 +42,18 @@ function _print_usage() {
         echo " General Options:"
         echo ""
         echo "  -h, --help                  Display this help, then exit."
+        echo "  --zap ZAP                   Use the zap program ZAP as he zap executable to "
```
Comment (bzbarsky-apple) [inline scripts/tools/zap/run_zaptool.sh:45]:
```suggestion
        echo "  --zap ZAP                   Use the zap program ZAP as the zap executable to "
``

File: scripts/tools/zap/run_zaptool.sh [hunk 2]
```diff
@@ -39,6 +42,18 @@ function _print_usage() {
...
+        echo "                              run (default: $ZAP_CMD_DEFAULT)."
+        echo ""
+        echo " Some influential environment variables:"
```
Comment (bzbarsky-apple) [inline scripts/tools/zap/run_zaptool.sh:48]: The key part is this: if either of these is used the `--zap` option will be ignored, right?  We should clearly document that interaction.

File: scripts/tools/zap/run_zaptool.sh (hunk 1 above)
Comment (gerickson) [reply scripts/tools/zap/run_zaptool.sh]: Good catch, thanks. Addressed and pushed.

File: scripts/tools/zap/run_zaptool.sh (hunk 2 above)
Comment (gerickson) [reply scripts/tools/zap/run_zaptool.sh:48]: Good feedback, thanks. Addressed and pushed.

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**